### PR TITLE
sanity as sub rules DOC-426

### PR DESCRIPTION
**Pull request by urikirsh**
Before requesting review:
 - Make sure CI is passing
   - Spell check failure may require adding backticks around code or updating `spelling_wordlist.txt`
   - See `README.md` for information about style and markdown syntax
   - If the CI Details link gives a 404, you need to log in to readthedocs.com

Jira ticket: https://certora.atlassian.net/browse/DOC-426
Link to generated documentation: https://certora-certora-prover-documentation--281.com.readthedocs.build/en/281/docs/prover/checking/sanity.html

### CODE REVIEW

**Comment by urikirsh** [comment]
Waiting for https://certora.atlassian.net/browse/CERT-6765

File: docs/prover/checking/sanity.md [hunk 1]
```diff
@@ -183,21 +189,27 @@ The assertion structure check tries to prove some complex logical statements by
 breaking them into simpler parts.  The following situations are reported by the
 assertion structure check:
 
-* `assert p => q;` is reported as a sanity violation if `p` is false whenever the
-  assertion is reached (in which case the simpler assertion `assert !p;` more
-  clearly describes the situation), or if `q` is always true (in which case
-  `assert q;` is a clearer alternative).
+1. `assert p => q;` is reported as a sanity violation if whenever the assertion is reached:
```
Comment (johspaeth) [inline docs/prover/checking/sanity.md:192]:
```suggestion
1. `assert p => q;` is reported as a sanity violation whenever the assertion is reached:
```

File: docs/prover/checking/sanity.md [hunk 2]
```diff
@@ -183,21 +189,27 @@ The assertion structure check tries to prove some complex logical statements by
...
+  * 1. `p` is always false. In this case, `q` is never checked. A simpler way to write this assertion is `assert !p;`. The node named `assertion_left_operand_check_<ID>` will have a yellow icon.
+
+  * 2. `p` is always true. A simpler way to write this assertion is `assert q;`. The node named `assertion_left_operand_check_<ID>` will have a yellow icon.
 
-* `assert p <=> q;` is reported as a sanity violation if either `p` and `q` are
-  both true whenever the assertion is reached (in which case the simpler
-  assertions `assert p; assert q;` more clearly describe the situation), or if
-  neither `p` nor `q` are ever true (in which case `assert !p; assert !q;` is a
-  clearer alternative).
+  * 3. `q` is always true. A simpler way to write this assertion is `assert q;`. The node named `assertion_right_operand_check_<ID>`.
```
Comment (johspaeth) [inline docs/prover/checking/sanity.md:198]:
```suggestion
  * 3. `q` is always true. A simpler way to write this assertion is `assert p;`. The node named `assertion_right_operand_check_<ID>`.
```

File: docs/prover/checking/sanity.md [hunk 3]
```diff
@@ -114,6 +112,10 @@ rule tautology {
 Since every `uint` satisfies the assertion, the assertion is tautological, which
 may indicate an error in the specification.
 
+The tautology check will add a node per `assert` or `satisfy` statement to each rule.  The nodes are named with the prefix `assert_not_tautological_<ID>`.  For example, the rule report for the above `tautology` rule will look like this:
```
Comment (johspaeth) [inline docs/prover/checking/sanity.md:115]: nit - `<ID>` isn't an ID, but the `<line number>_<start_column>` of the expression. tioli if you think this detail is required in the docs.

**Review by johspaeth — APPROVED** [review]
LGTM, minor nits. Thanks for supporting in updating the docs!

File: docs/prover/checking/sanity.md (hunk 3 above)
Comment (urikirsh) [reply docs/prover/checking/sanity.md]: Changed everywhere to `<LINE>_<COL>` as a middle ground that explains it but is still shorter

File: docs/prover/checking/sanity.md (hunk 2 above)
Comment (urikirsh) [reply docs/prover/checking/sanity.md]: good catch

**Pull request merged**